### DISCUSSION TITLE
How can I deploy pocketbase on a VPS with other projects?

### DISCUSSION THREAD

**Discussion by josema701** [pocketbase/pocketbase#176] [Q&A] [answered]
I want to deploy Pocketbase in a digitalocean droplet with nginx, the problem is that after doing what the documentation says I have a: 502 Bad Gateway.
And when I execute ./pocketbase serve, it tells me that the pocketbase command does not exist

**Comment by ganigeorgiev** [comment, marked as answer]
Could you provide a little more info about your setup - logs, nginx config(s)?
You could also try running `nginx -t` to ensure that there aren't any syntax errors in the configurations.

About the missing command - you have to provide the full path to the PocketBase executable if your working directory is not where the `pocketbase` file is located, eg. `/path/to/pocketbase serve`. If that doesn't work, make sure that the file has permissions to be executed, eg. `chmod +x /path/to/pocketbase`.

**Reply by josema701** [reply]
![image](https://user-images.githubusercontent.com/47338136/180036690-18b3c588-85d8-4a1f-960e-a0f2530ccdbb.png)

result running nginx -t

and Nginx config:
![image](https://user-images.githubusercontent.com/47338136/180037392-1f9f4ce3-d943-433f-9e1f-4b64ccd529fd.png)

**Reply by ganigeorgiev** [reply]
Is the PocketBase service running on the 8090 port? Aka.:
```
/path/to/pocketbase serve
```

_You can also create a systemd for the `pocketbase` executable so that it can start automatically on server restart/process kill - https://pocketbase.io/docs/going-to-production/#create-systemd-service_

**Reply by josema701** [reply]
![image](https://user-images.githubusercontent.com/47338136/180039498-2ca7ea8b-430c-42b2-955d-a26ade966b1c.png)
its ok?

**Reply by ganigeorgiev** [reply]
If you are using nginx and don't want to change the default 8090 port, then the `--http` and `--https` flags should be removed, or in the above screenshot your `ExecStart` should be just:
```
ExecStart = /var/www/pocketbase/pocketbase serve
```

**Reply by josema701** [reply]
![image](https://user-images.githubusercontent.com/47338136/180055505-48403e28-0cb6-4c27-9113-daf0a0d8077c.png)

how do i fix this error?:

![image](https://user-images.githubusercontent.com/47338136/180055768-45a62c37-eaf8-4b8e-a1d6-1e5bc1edd298.png)

**Reply by ganigeorgiev** [reply]
What is the result of directly executing the binary - `/var/www/pocketbase/pocketbase serve`?

**Reply by josema701** [reply]
![image](https://user-images.githubusercontent.com/47338136/180061877-cf38870c-073c-4080-be4e-26e49be6b992.png)

executing it runs fine, but in service it doesn't

**Reply by josema701** [reply]
It works now, I deleted everything and did it again, and now it works.

Thanks for the help.

**Comment by joesandiroz2** [comment]
not work for me

root@androidSamsung:/home/buntu/webdood/d-scanner# service pocketbase status
● pocketbase.service - pocketbase
     Loaded: loaded (/usr/lib/systemd/system/pocketbase.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Mon 2025-02-10 11:15:24 WIB; 742ms ago
    Process: 21249 ExecStart=/home/buntu/webdood/pocketbase/pocketbase serve (code=exited, status=209/S>
   Main PID: 21249 (code=exited, status=209/STDOUT)
        CPU: 2ms

**Comment by joesandiroz2** [comment]
here pocketbase.service file

[Unit]
Description = pocketbase

[Service]
Type             = simple
User             = root
Group            = root
LimitNOFILE      = 4096
Restart          = always
RestartSec       = 5s
StandardOutput   = append:/root/pb/std.log
StandardError    = append:/root/pb/std.log
WorkingDirectory = /home/buntu/webdood/pocketbase/
ExecStart        = /home/buntu/webdood/pocketbase/pocketbase serve


[Install]
WantedBy = multi-user.target


and here result for run 
root@androidSamsung:/home/buntu/webdood/grafindo-scanner# /home/buntu/webdood/pocketbase/pocketbase serve
2025/02/10 11:17:42 Server started at http://127.0.0.1:8090
├─ REST API:  http://127.0.0.1:8090/api/
└─ Dashboard: http://127.0.0.1:8090/_/


but still not work

root@androidSamsung:/home/buntu/webdood/grafindo-scanner# service pocketbase status
● pocketbase.service - pocketbase
     Loaded: loaded (/usr/lib/systemd/system/pocketbase.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Mon 2025-02-10 11:17:56 WIB; 482ms ago
    Process: 21384 ExecStart=/home/buntu/webdood/pocketbase/pocketbase serve (code=exited, status=209/S>
   Main PID: 21384 (code=exited, status=209/STDOUT)
        CPU: 793us
lines 1-6/6 (END)

**Reply by ganigeorgiev** [reply]
@joesandiroz2 Please don't spam! This is a very old discussion and a lot of things has changed.

I'm not sure what is the issue in your case but as mentioned in the above comments - make sure that the stdout/stderr logs dir (in your case `/root/pb`) exists.

If you need further help, open a new Q&A discussion but try to provide more details about your use case and setup to understand what you are trying to do.

If you are new to web development and/or don't know how to setup a VPS, then please follow exactly the steps described in https://pocketbase.io/docs/going-to-production/#minimal-setup.
